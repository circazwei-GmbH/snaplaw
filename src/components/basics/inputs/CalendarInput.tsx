import React, { useState } from "react";
import {
  GestureResponderEvent,
  Pressable,
  StyleProp,
  StyleSheet,
  Text,
  TextStyle,
  View,
} from "react-native";
import VerticalDivider from "../dividers/VerticalDivieder";
import DefaultText from "../typography/DefaultText";
import { MaterialCommunityIcons } from "@expo/vector-icons";
import DatePickerModalCustom from "../../features/Modals/DatePickerModalCustom";
import dayjs from "dayjs";

type CalendarInputProps = {
  style?: StyleProp<TextStyle>;
  date: string;
  dateHandler: (date: string) => void;
  placeholder?: string;
  errorMessage?: string;
};

export default function CalendarInput({
  style,
  date,
  dateHandler,
  placeholder = "",
  errorMessage,
}: CalendarInputProps) {
  const [datePickerOpened, setDatePickerOpened] = useState(false);

  const onConfirmDate = (date: string) => {
    dateHandler(date);
    setDatePickerOpened(false);
  };

  return (
    <>
      <View
        style={[styles.container, style, errorMessage ? styles.error : null]}
      >
        <View style={styles.signContainer}>
          {dayjs(date).isValid() ? (
            <DefaultText
              text={dayjs(date).format("DD.MM.YYYY")}
              testID="CalendarInputDate"
            />
          ) : (
            <Text testID="CalendarInputDate" style={styles.placeholder}>{placeholder}</Text>
          )}
        </View>
        <View style={styles.rightPart}>
          <View style={styles.dividerContainer}>
            <VerticalDivider />
          </View>
          <Pressable
            onPress={() => setDatePickerOpened(true)}
            style={styles.iconContainer}
            testID="DataPickerPressabelAreaID"
          >
            <MaterialCommunityIcons
              name="calendar-month-outline"
              size={22}
              color="#668395"
            />
          </Pressable>
        </View>
        <DatePickerModalCustom
          onConfirmDate={onConfirmDate}
          onCancelDate={() => setDatePickerOpened(false)}
          datePickerOpened={datePickerOpened}
          setDatePickerOpened={setDatePickerOpened}
          setedDate={date}
        />
      </View>
      <Text
        style={[styles.errorText, errorMessage ? null : styles.displayNone]}
      >
        {errorMessage}
      </Text>
    </>
  );
}

const styles = StyleSheet.create({
  container: {
    flexDirection: "row",
    backgroundColor: "#EFF7FD",
    height: 44,
    borderRadius: 10,
    justifyContent: "space-between",
    alignItems: "center",
  },
  rightPart: {
    flexDirection: "row",
    alignItems: "center",
  },
  iconContainer: {
    paddingHorizontal: 29,
  },
  dividerContainer: {
    paddingVertical: 7,
  },
  signContainer: {
    width: "70%",
    alignItems: "center",
    justifyContent: "center",
  },
  image: {
    width: "100%",
    height: "100%",
  },
  placeholder: {
    color: "#909090",
    fontSize: 15,
    width: "100%",
    paddingLeft: 16,
  },
  error: {
    borderWidth: 1,
    borderColor: "#FA7171",
  },
  errorText: {
    paddingTop: 5,
    color: "#FA7171",
  },
  displayNone: {
    display: "none",
  },
});
