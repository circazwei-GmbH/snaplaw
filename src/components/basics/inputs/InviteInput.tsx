import React from 'react'
import {GestureResponderEvent, Pressable, StyleProp, StyleSheet, TextStyle, View} from "react-native";
import VerticalDivider from "../dividers/VerticalDivieder";
import DefaultText from "../typography/DefaultText";
import { Feather } from '@expo/vector-icons';

type InviteInputProps = {
    style?: StyleProp<TextStyle>,
    invitedName: string,
    inviteHandler: (event: GestureResponderEvent) => void
}

export default function InviteInput({style, invitedName, inviteHandler}: InviteInputProps) {
    return (
        <View style={[styles.container, style]}>
            <View style={styles.signContainer}>
                <DefaultText text={invitedName} />
            </View>
            <View style={styles.rightPart}>
                <View style={styles.dividerContainer}>
                    <VerticalDivider />
                </View>
                <Pressable onPress={inviteHandler}>
                    <View style={styles.iconContainer} testID="InvitePressabelAreaID">
                        <Feather name="user-plus" size={22} color="#668395" />
                    </View>
                </Pressable>
            </View>
        </View>
    )
}

const styles = StyleSheet.create({
    container: {
        flexDirection: "row",
        backgroundColor: '#EFF7FD',
        height: 44,
        borderRadius: 10,
        justifyContent: "space-between",
        alignItems: "center"
    },
    rightPart: {
        flexDirection: "row",
        alignItems: 'center'
    },
    iconContainer: {
        paddingHorizontal: 29
    },
    dividerContainer: {
        paddingVertical: 7
    },
    signContainer: {
        width: '70%',
        alignItems: "center",
        justifyContent: "center"
    },
    image: {
        width: '100%',
        height: '100%',
    }
})