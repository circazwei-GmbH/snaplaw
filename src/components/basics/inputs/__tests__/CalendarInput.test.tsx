import React from "react";
import { fireEvent, render } from "@testing-library/react-native";
import InviteInput from "../InviteInput";
import CalendarInput from "../CalendarInput";

describe("CalensarInput", () => {
  it("Should be dispay name", () => {
    const { getByText } = render(
      <CalendarInput date={`${new Date()}`} placeholder="Date" dateHandler={()=>{}}/>
    );
    expect(getByText("Date*")).toBeTruthy();
  });
  it("Should call handler", () => {
    const { toJSON, getByText, getByTestId } = render(
      <CalendarInput date={`${new Date()}`} placeholder="Date" dateHandler={()=>{}} />
    );
    fireEvent.press(getByTestId("DataPickerPressabelAreaID"));
    fireEvent.press(getByText("menu.cancel"));
    expect(toJSON()).toMatchSnapshot();
  });
});
