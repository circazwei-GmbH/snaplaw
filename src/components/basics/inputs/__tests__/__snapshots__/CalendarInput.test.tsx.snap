// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`CalensarInput Should call handler 1`] = `
Array [
  <Text
    style={
      Object {
        "color": "#1696E2",
        "fontSize": 14,
        "lineHeight": 22,
      }
    }
  >
    Date
    <Text
      style={
        Object {
          "color": "#FA7171",
        }
      }
    >
      *
    </Text>
  </Text>,
  <View
    style={
      Array [
        Object {
          "alignItems": "center",
          "backgroundColor": "#EFF7FD",
          "borderRadius": 10,
          "flexDirection": "row",
          "height": 44,
          "justifyContent": "space-between",
        },
        undefined,
        null,
        null,
      ]
    }
  >
    <View
      style={
        Object {
          "alignItems": "center",
          "justifyContent": "center",
          "width": "70%",
        }
      }
    >
      <Text
        style={
          Array [
            Object {
              "color": "#202020",
              "fontFamily": "P",
              "fontSize": 17,
            },
            undefined,
          ]
        }
        testID="CalendarInputDate"
      >
        15.11.2021
      </Text>
    </View>
    <View
      style={
        Object {
          "alignItems": "center",
          "flexDirection": "row",
        }
      }
    >
      <View
        style={
          Object {
            "paddingVertical": 7,
          }
        }
      >
        <View
          style={
            Object {
              "backgroundColor": "#919191",
              "height": "100%",
              "width": 0.5,
            }
          }
        />
      </View>
      <View
        accessible={true}
        focusable={true}
        onBlur={[Function]}
        onClick={[Function]}
        onFocus={[Function]}
        onResponderGrant={[Function]}
        onResponderMove={[Function]}
        onResponderRelease={[Function]}
        onResponderTerminate={[Function]}
        onResponderTerminationRequest={[Function]}
        onStartShouldSetResponder={[Function]}
        style={
          Object {
            "paddingHorizontal": 29,
          }
        }
        testID="DataPickerPressabelAreaID"
      >
        <Text />
      </View>
    </View>
    <Modal
      animationType="none"
      hardwareAccelerated={false}
      transparent={true}
      visible={false}
    >
      <View
        style={
          Array [
            Object {
              "alignItems": "center",
              "backgroundColor": "rgba(0,0,0,0.7)",
              "flex": 1,
              "justifyContent": "flex-end",
            },
            Object {
              "justifyContent": "center",
            },
          ]
        }
      >
        <View
          style={
            Object {
              "alignItems": "flex-end",
              "backgroundColor": "#fff",
              "padding": 20,
            }
          }
        >
          <View
            style={
              Object {
                "flexDirection": "row",
                "marginTop": 15,
              }
            }
          >
            <View
              accessible={true}
              focusable={true}
              onClick={[Function]}
              onResponderGrant={[Function]}
              onResponderMove={[Function]}
              onResponderRelease={[Function]}
              onResponderTerminate={[Function]}
              onResponderTerminationRequest={[Function]}
              onStartShouldSetResponder={[Function]}
              style={
                Object {
                  "opacity": 1,
                }
              }
            >
              <View
                style={
                  Object {
                    "marginLeft": 20,
                  }
                }
              >
                <Text
                  style={
                    Object {
                      "fontSize": 16,
                      "textTransform": "uppercase",
                    }
                  }
                >
                  menu.cancel
                </Text>
              </View>
            </View>
            <View
              accessible={true}
              focusable={true}
              onClick={[Function]}
              onResponderGrant={[Function]}
              onResponderMove={[Function]}
              onResponderRelease={[Function]}
              onResponderTerminate={[Function]}
              onResponderTerminationRequest={[Function]}
              onStartShouldSetResponder={[Function]}
              style={
                Object {
                  "opacity": 1,
                }
              }
              testID="ConfirmDate"
            >
              <View
                style={
                  Object {
                    "marginLeft": 20,
                  }
                }
              >
                <Text
                  style={
                    Object {
                      "fontSize": 16,
                      "textTransform": "uppercase",
                    }
                  }
                >
                  menu.confirm
                </Text>
              </View>
            </View>
          </View>
        </View>
      </View>
    </Modal>
  </View>,
  <Text
    style={
      Array [
        Object {
          "color": "#FA7171",
          "paddingTop": 5,
        },
        Object {
          "display": "none",
        },
      ]
    }
  />,
]
`;
